lib/local_system.rb

```diff
@@ -29,17 +29,32 @@ def os
       @@os
     end
 
-    def validate_existence_of_package(package)
-      begin
-        Cheetah.run("rpm", "-q", package)
-      rescue
-        needed_module = os.module_required_by_package(package)
-        if needed_module
-          raise(Machinery::Errors::MissingRequirement.new("You need the package '#{package}' from module '#{needed_module}'. You can install it as follows:\n" \
-            "If you haven't selected the module '#{needed_module}' before, run `yast2 scc` and choose 'Select Extensions' and activate '#{needed_module}'.\nRun `zypper install #{package}` to install the package."))
-        else
-          raise(Machinery::Errors::MissingRequirement.new("You need the package '#{package}'. You can install it by running `zypper install #{package}`"))
+    def validate_existence_of_packages(packages)
+      missing_packages = []
+      needed_modules = []
+      error_string = ""
+
+          packages.each do |package|
+            begin
+              Cheetah.run("rpm", "-q", package)
+            rescue Cheetah::ExecutionFailed
+              missing_packages << package
+              needed_module = os.module_required_by_package(package)
+              if needed_module
+                needed_modules << package
+                if needed_module
+                  raise(Machinery::Errors::MissingRequirement.new("You need the package '#{package}' from module '#{needed_module}'. You can install it as follows:\n" \
+                    "If you haven't selected the module '#{needed_module}' before, run `yast2 scc` and choose 'Select Extensions' and activate '#{needed_module}'.\nRun `zypper install #{package}` to install the package.\n\n"))
+                end
+              end
+            end
+          end
+
+      if !missing_packages.empty?
+        missing_packages.each do |missing_package|
+          error_string += "You need the package '#{missing_package}'. You can install it by running `zypper install #{missing_package}`\n"
```

Comment:
It would be better to put out one message for installation of all packages instead of a separate one for each package.

It also would be good to have a unit test, which checks the error message. This would also make it easier to see what the code does, when the unit test demonstrates how the expected output should look like.